analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,394 @@
+
+```{r}
+# Read in the annotated gene level CN file
+consensus_seg_autosomes_df <-
+  read_tsv(file.path(results_dir, "consensus_seg_annotated_cn_autosomes.tsv.gz"))
+```
+
+## Define most focal units
+
+### Determine chromosome arm status
+
+```{r}
+# Use our custom function to make the arm status calls
+arm_status_count <- status_majority_caller(
+  consensus_seg_cytoband_status_df,
+  "chromosome_arm",
+  status_threshold
+)
+
+# Display table
+arm_status_count %>%
+  group_by(dominant_status)
+```
+
+These are the chromosome arms that have not been defined as gain or loss -- we want to define their cytoband/gene-level status
+
+```{r}
+# Let's get a vector of the neutral arms
+neutral_arms <- arm_status_count %>%
+  filter(dominant_status == "neutral") %>%
+  dplyr::pull("chromosome_arm")
+```
+
+### Determine cytoband status
+
+We want to include cytoband and gene-level calls for chromosome arms that have not been defined as a gain or loss to make the cytoband-level majority calls.
+
+```{r}
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than 90% of the total counts are
+# for that particular status
+cytoband_status_count <- consensus_seg_cytoband_status_df %>%
+  # Filter the annotated CN data to include only neutral chromosome arms
+  filter(chromosome_arm %in% neutral_arms) %>%
+  select(
+    Kids_First_Biospecimen_ID,
+    cytoband,
+    loss_fraction,
+    gain_fraction,
+    callable_fraction,
+    dominant_status = dominant_cytoband_status
+  ) %>%
+  mutate(
+    dominant_status = case_when(
+      callable_fraction < (1 - uncallable_threshold) ~ "uncallable",
+      loss_fraction > status_threshold ~ "loss",
+      gain_fraction > status_threshold ~ "gain",
+      loss_fraction + gain_fraction > status_threshold ~ "unstable",
+      TRUE ~ "neutral"
+    )
+  )
+
+# Display table
+cytoband_status_count %>%
+  group_by(dominant_status)
+```
+
+### Determine gene-level status
+
+```{r}
+# These are the cytobands that have not been defined as gain or loss --
+# we want to define their gene-level status
+neutral_cytobands <- cytoband_status_count %>%
+  filter(dominant_status %in% c("unstable", "neutral")) %>%
+  dplyr::pull("cytoband")
+
+# Filter the annotated CN data to include only these cytobands
+neutral_status_cytoband_df <- consensus_seg_cytoband_status_df %>%
+  filter(cytoband %in% neutral_cytobands)
+
+# Join gene-level data
+combined_neutral_status_cytoband_df <- neutral_status_cytoband_df %>%
+  left_join(
+    consensus_seg_autosomes_df,
+    by = c("Kids_First_Biospecimen_ID" = "biospecimen_id", "cytoband")
+  )
+```
+
+```{r}
+if (!(running_in_ci)) {
+  # Now count the gene-level calls (for each status call) and define
+  # the gene as that status if more than 50% of the total counts are
+  # for that particular status
+  gene_status_count <- status_majority_caller(
+    combined_neutral_status_cytoband_df,
+    "gene_symbol",
+    status_threshold
+  )
```

Comment:
I don't understand what this is meant to do. Each sample should only have one call for each gene, right?

Reply:
You are correct, it was redundant and I believe I resolved this in the last commit.

Reply:
You are correct, it was redundant and I believe I resolved this in the last commit.